Android/ThirdPartyAuth/README.MD

```diff
@@ -0,0 +1,45 @@
+# ThirdPartyAuth
+This sample demonstrates using an iTwin service app so that an iTwin Mobile app can make use of a 3rd party authentication provider. It uses [auth0](https://auth0.com/) for user authentication. The sample uses Jetpack Compose UI to authenticate against an auth0 account and obtain an ID token. It then uses that ID token as the authentication when communicating with a demo iTwin token provider server running on the developer's computer. The token provider server validates that the token is from the proper auth0 domain, and then uses a client secret that was created during the registration of the iTwin service app to generate an iTwin access token.
+
+If you want to run the sample, you will need to set up an auth0 account and configure it to work with this sample. However, while this sample uses auth0, the 3rd party authentication workflow should work the same with any authentication provider. Specifically:
+
+1. Have the user sign in and get a token.
+1. Automatically refresh this token as needed.
+1. Use this token in the `Authentication` header sent to a custom iTwin access token provider.
+1. Create a web service app that runs on a server somewhere.
+1. Have this app validate the incoming token to decide whether the user is authorized, and if so to what extent.
+1. Return an iTwin service access token based on your authentication of the user.
+
+## Setup
+
+1. Follow the instructions in the [Android Readme](../README.md).
+1. Create an client on [auth0](https://auth0.com/).
```

Comment:
The auth0 setup instructions should be refactored into a separate .md file that is linked to here and also linked to in the iOS setup instructions. I would suggest using ../../cross-platform/token-server/AUTH0.md.